### DISCUSSION TITLE
[5.0][6.0] Use enums and build them like the current CustomFieldBehavior for sections/globalsets/fields

### DISCUSSION THREAD

**Discussion by Anubarak** [craftcms/cms#11421] [Ideas]
With the introduction of [enums ](https://www.zend.com/blog/php-8-1#enums) in PHP 8.1 we could actually go all the way to native enum support in the future.

For example you could integrate a `Section`, `GlobalSet`, `Category` and so on enum and create dynamic classes on the fly, when those datatypes are changed.

All functions could then be called via 
```PHP
// Sections.php
public function getSectionByHandle(SectionHandle $handle): ?Section

// another file - auto generated after save
enum SectionHandle 
{
    case Pages;
    case News;
    case Whatever;
}
```
or in twig
```twig
{# let's pretent you inject this enum into Twig and PHPStorm does the rest to autocomplete them fine #}
{% set entries = craft.entries.section(SectionHandle.pages).all() %}
```

**Comment by brandonkelly** [comment]
Agree that would be awesome :)
